src/ast/passes/codegen_llvm.cpp

```diff
@@ -3487,6 +3515,8 @@ void CodegenLLVM::probereadDatastructElem(Value *src_data,
     // below triggering an assert in LLVM
     src_data = b_.CreateIntCast(src_data, b_.getInt64Ty(), false);
   }
+  else if (src_ty->isPointerTy())
```

Comment:
Note to other readers: 
This worked before this PR because the values were being passed around as integers. Now that this PR makes some of the values retain pointer types, we need explicit casts of the inputs to proberead.